### ISSUE TITLE
Why VRAM is increasing while training?

### ISSUE THREAD

**Issue by goreng2** [ultralytics/ultralytics#20750]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hi Ultralytics team,

I'm currently training a `YOLOv11n` model with my custom dataset. I'm using Ultralytics version `8.3.107`.

My training script is as follows:
```
    results = yolo.train(
        data=dataset,
        epochs=1000,
        batch=16,
        imgsz=320,
        device=0,
        save=True,
    )
```

I've noticed that my GPU VRAM usage steadily increases throughout the training process. For example, it starts at around `2GB` and progressively climbs, reaching `9.81GB` by the end of training. My GPU is a `NVIDIA A100 80GB`.

My questions are:
1. Is this continuous increase in VRAM usage during training a normal behavior for YOLOv11 (or Ultralytics in general)?
2. Or could this indicate a potential memory leak or an issue that I should be concerned about?

For context, I am not using the cache option (e.g., `cache=True` or `cache=disk`).

Any insights or recommendations would be greatly appreciated. Thank you!

### Additional

![Image](https://github.com/user-attachments/assets/4f4708fe-175a-46b9-b16f-6bb3c17cfe9a)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @goreng2, thank you for reaching out and providing detailed information about your training setup with YOLO11n! 🚀 This is an automated response to help get you started—an Ultralytics engineer will also review your issue and assist you soon.

We recommend checking our [Docs](https://docs.ultralytics.com/) for detailed [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples, where common questions about training and resource usage are often addressed.

If this is a 🐛 Bug Report (such as a potential memory leak), please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us investigate further.

If you are seeking advice or troubleshooting for custom training, please continue to include information like dataset details, training logs, and screenshots, and ensure you are following the [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community to connect in real time or discuss further:
- [Discord](https://discord.com/invite/ultralytics) 🎧 for live chat
- [Discourse](https://community.ultralytics.com/) for in-depth discussions
- [Subreddit](https://reddit.com/r/Ultralytics) to share and learn with others

## Upgrade

To ensure you benefit from the latest bug fixes and features, please upgrade to the newest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO runs in these up-to-date verified environments (all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

A green badge means all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are passing. CI tests check YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you for your detailed report and patience!

**Comment by Y-T-G** [comment]
I think it's normal. It's the amount of memory reserved by PyTorch. It will be freed if needed automatically.
